quic/transport/ngtcp2/connection/openstate.nim

```diff
@@ -0,0 +1,67 @@
+import pkg/chronos
+import ../../../udp/datagram
+import ../../quicconnection
+import ../../connectionid
+import ../../stream
+import ../connection
+import ../streams
+import ./closingstate
+import ./drainingstate
+import ./disconnectingstate
+
+type
+  OpenConnection* = ref object of ConnectionState
+    quicConnection: QuicConnection
+    ngtcp2Connection: Ngtcp2Connection
+
+proc newOpenConnection*(ngtcp2Connection: Ngtcp2Connection): OpenConnection =
+  OpenConnection(ngtcp2Connection: ngtcp2Connection)
+
+{.push locks: "unknown".}
+
+method enter(state: OpenConnection, connection: QuicConnection) =
+  state.quicConnection = connection
```

Comment:
similar assert

Reply:
I added protection against state reentry in 923ad8d.